boards/stk3700/dist/renode/board.resc

```diff
@@ -0,0 +1,25 @@
+:name: STK3700
+:description: Silicon Labs STK3700 starter kit
+
+$name?="STK3700"
+
+using sysbus
+mach create $name
+
+machine LoadPlatformDescription $RIOTCPU/efm32/families/efm32gg/dist/renode/efm32gg990f1024.repl
+machine LoadPlatformDescription $RIOTBOARD/stk3700/dist/renode/stk3700.repl
```

Comment:
Ideally, yes. But they are not always compatible. Also, not all EFMs are in there.

Reply:
I agree that they should be in upstream. But for this PR to test, you need them currently ;-)

Reply:
About the stk3700, this PR adds the board .repl:

```
bc: Miscellaneous.LED @ gpioPort 87

led0_pin: Miscellaneous.LED @ gpioPort 66
led1_pin: Miscellaneous.LED @ gpioPort 67

pb0_pin: Miscellaneous.Button @ gpioPort 25
pb1_pin: Miscellaneous.Button @ gpioPort 26
```

upstream:

```
using "platforms/cpus/silabs/efm32xg/efm32gg995.repl"

led0: Miscellaneous.LED @ gpioPort 66
led1: Miscellaneous.LED @ gpioPort 67

button0: Miscellaneous.Button @ gpioPort 64
button1: Miscellaneous.Button @ gpioPort 65

deviceInformation:
    deviceNumber: 990
```

How is it possible that they are so different ? (different cpus, some pin mappings are different, led0_pin -> led0, etc).

> not all EFMs are in there.

So why not add them in renode directly ?

Reply:
@aabadie They are in[ renode directly](https://github.com/renode/renode/tree/master/platforms/cpus/silabs).
You might want to have different .resc and .repl files though. From my perspective there are a couple of reason not to use the renode definitions. I use a local installation of renode. Providing the global files doesn't work there, since the [relative path has some issues](https://github.com/renode/renode/issues/127). You also want to change the definition in some cases. For example to expose the GDB interface or things like this. Renode also changed some things over time. I couldn't even use the emulation with the recent version. Some register entries for the nRF52840 are missing, for example.
So, we might even want to consider to only support certain versions, to make sure that it still works.

Reply:
I was not talking about the .resc files as they depends on RIOT specific variables (the elffile at least). I agree that eventually the repl could be different because RIOT doesn't use the same default configuration. But for the CPU repl, let me share some doubts because the CPU definitions are not supposed to change (the register addresses, etc).
The problem with relative paths seems to be an issue on their side. TBH it's super simple to install Renode using the package manager of your system (the latest release provides OSX packages, .deb and .rpm). If you decide to build it locally, I guess setting the PATH variable accordingly shouldn't be an issue.

You are also mentioning the versions. I looked in detail at how they are handled by this PR and I don't like it: with this PR you would have to use 1.12 which is not release yet and although 1.11 still works with cc2538dk. I think this should be handed differently, with the boards having renode configuration defining a variable with the minimum version required.

I've also been working on improving the renode integration and could came up with something only using Makefiles (no need for a temporary resc file in the application directory) and I'm also able to use the debugger (but start should be called from the board.resc).
If you are interested, I can PR this work.

Reply:
I’m interested in your approach, @aabadie. Can you maybe already share a branch?

Reply:
Here it is: https://github.com/aabadie/RIOT/tree/pr/tools/renode_refactor

You have to start renode in debugger mode using `make BOARD=cc2538dk -C examples/hello-world all emulate-debug-server` and then in another terminal, just use `make BOARD=cc2538dk -C examples/hello-world debug` to connect to the debug server.
If we want to have something fully automatic (make debug also start the debug server in background), we would have to use a shell script (similar to openocd).

Reply:
"TBH it's super simple to install Renode using the package manager of your system (the latest release provides OSX packages, .deb and .rpm)."
That is true. But you can say this for every other tool as well. In sum you end up with a lot of different packages you need to install and a lot of configuration you need to change. I want to make it more user friendly and therefore want to only have one snap package with everything in it. To reduce complexity as much as possible. Eventually I want to have something as simple as the Arduino IDE. So that is why this is an important point for me.

Reply:
> If we want to have something fully automatic (make debug also start the debug server in background), we would have to use a shell script (similar to openocd).

I could manage to have a better integration of emulate with debug and debug-server targets. Now with the branch mentioned above, it's possible to run `make BOARD=cc2538dk -C examples/hello-world emulate debug` to start renode with a GDB server and automatically attach to it using gdb, like with openocd.

Reply:
Finally went for a PR: #15490